router/functionHandler.go

```diff
@@ -52,11 +64,12 @@ func (fh *functionHandler) getServiceForFunction() (*url.URL, error) {
 type RetryingRoundTripper struct {
 	maxRetries    int
 	initalTimeout time.Duration
+	transport     http.Transport
 }
 
 func (rrt RetryingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	timeout := rrt.initalTimeout
-	transport := http.DefaultTransport.(*http.Transport)
+	transport := &rrt.transport
```

Comment:
The default transport will send all the traffic to the same backend pod via the service url, though there are several pods behind the service.